### ISSUE TITLE
Ubuntu 14.04 Error using DrawText: Failed to load external drawtext plugin

### ISSUE THREAD

**Issue by hanneshberlin** [Psychtoolbox-3/Psychtoolbox-3#400]
I cannot run "DrawText" function. I tried everything I could find online including reinstalling PTB, nothing helped. Any ideas? The 'help DrawTextPlugin' didnt help either.

My system and setup:
OS: Ubuntu 14.04 LTS
Matlab: R2012b (8.0.0.783) 64-bit
PTB version: 3.0.14 - Flavor: Debian package - psychtoolbox-3 (3.0.14.20170103+git6-g605ff5c.dfsg1-1~nd14.04+1)

I am getting this error:

PTB-WARNING: DrawText: Failed to load external drawtext plugin 'libptbdrawtext_ftgl64.so.1' [libptbdrawtext_ftgl64.so.1: Cannot open Shared-Object-File: File or folder not found]. Reverting to legacy text renderer.
PTB-WARNING: DrawText: Functionality of Screen('DrawText') and Screen('TextBounds') may be limited and text quality may be impaired.
PTB-WARNING: DrawText: Type 'help DrawTextPlugin' at the command prompt to receive instructions for troubleshooting.

Failed to load X11 font with name Arial.

Available X11 fonts are:

[long list]

Failed to load X11 font with name -*-arial-regular-r-*--24-*-*-*.
Try a Screen('TextFont') name according to one of the listed available fonts above.

Error in function DrawText: 	Usage error
Couldn't select the requested font with the requested font settings from X11 system!

The 'libptbdrawtext_ftgl64.so.1' is linked at "/usr/share/matlab/site/m/psychtoolbox-3/psychtoolbox-3/PsychBasic/PsychPlugins/" to "../../../../lib/psychtoolbox-3/PsychBasic/PsychPlugins/libptbdrawtext_ftgl.so.1"

The font Arial is not within the X11 fonts but in truetype. Is that the problem? What could be a workaround?

Thank you very much in advance for your help!

**Comment by hanneshberlin** [comment]
Additional information

Lib-versions:
fontconfig 2.11.0-0ubuntu4.2
libfreetype6 2.5.2-1ubuntu2.5
freeglut3 2.8.1-1

Might this be part of the problem?
>> PsychtoolboxRoot

ans = /usr/share/matlab/site/m/psychtoolbox-3/

>> path

		MATLABPATH

	/usr/share/matlab/site/m/psychtoolbox-3
	/usr/share/matlab/site/m/psychtoolbox-3/PsychGamma
	/usr/share/matlab/site/m/psychtoolbox-3/PsychProbability
	/usr/share/matlab/site/m/psychtoolbox-3/PsychSound
	/usr/share/matlab/site/m/psychtoolbox-3/PsychSound/MOAL
	/usr/share/matlab/site/m/psychtoolbox-3/PsychSound/MOAL/source
	/usr/share/matlab/site/m/psychtoolbox-3/PsychSound/MOAL/core
	/usr/share/matlab/site/m/psychtoolbox-3/PsychSound/MOAL/wrap
	/usr/share/matlab/site/m/psychtoolbox-3/PsychAlphaBlending
	/usr/share/matlab/site/m/psychtoolbox-3/PsychStairCase
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOptics
	/usr/share/matlab/site/m/psychtoolbox-3/PsychVideoCapture
	/usr/share/matlab/site/m/psychtoolbox-3/PsychSignal
	/usr/share/matlab/site/m/psychtoolbox-3/PsychContributed
	/usr/share/matlab/site/m/psychtoolbox-3/PsychContributed/AutoBrightness
	/usr/share/matlab/site/m/psychtoolbox-3/PsychContributed/ArmArch
	/usr/share/matlab/site/m/psychtoolbox-3/PsychInitialize
	/usr/share/matlab/site/m/psychtoolbox-3/PsychJava
	/usr/share/matlab/site/m/psychtoolbox-3/PsychMatlabTests
	/usr/share/matlab/site/m/psychtoolbox-3/PsychCalDemoData
	/usr/share/matlab/site/m/psychtoolbox-3/PsychGPGPU
	/usr/share/matlab/site/m/psychtoolbox-3/Psychometric
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDocumentation
	/usr/share/matlab/site/m/psychtoolbox-3/PsychColorimetric
	/usr/share/matlab/site/m/psychtoolbox-3/PsychColorimetric/PsychMunsell
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/OpenGL4MatlabDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/OpenGL4MatlabDemos/GLSLDemoShaders
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/OpenGL4MatlabDemos/MorphDemo
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/MovieDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/SoundFiles
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/PsychExampleExperiments
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/PsychExampleExperiments/OldNewRecognition
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/PsychExampleExperiments/OldNewRecognition/stims
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/GPGPUDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/ECVP2013
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/PsychTutorials
	/usr/share/matlab/site/m/psychtoolbox-3/PsychDemos/AlphaImageDemo
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOneliners
	/usr/share/matlab/site/m/psychtoolbox-3/Quest
	/usr/share/matlab/site/m/psychtoolbox-3/PsychCal
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/Daq
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/Daq/Utilities
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/PR705Toolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/LinuxX11ExampleXorgConfs
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/LinuxDrivers
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/LinuxDrivers/NVidiaOptimus
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/PR670Toolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/ColorCal2
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/PR650Toolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/DatapixxToolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/DatapixxToolbox/DatapixxBasic
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/PsychVRToolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemo
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemo/ELCustomCalibration
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemo/AntiSaccade
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemo/change
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemo/pursuit
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemo/EyelinkFixationWindow
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/EyelinkShortDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkDemos/GazeContingentDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkBasic
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkTests
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/EyelinkToolbox/EyelinkOneLiners
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/iViewXToolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/iViewXToolbox/iviewxtests
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/iViewXToolbox/tcp_udp_ip
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/iViewXToolbox/cbase64
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/BrightSideDisplay
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/BrightSideDisplay/BSRuntimeLibs
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/PR655Toolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/PsychGamepad
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/BitsPlusToolbox
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/BitsPlusToolbox/BitsPlusDemos
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/BitsPlusToolbox/BitsPlusTests
	/usr/share/matlab/site/m/psychtoolbox-3/PsychHardware/FORP
	/usr/share/matlab/site/m/psychtoolbox-3/PsychAlpha
	/usr/share/matlab/site/m/psychtoolbox-3/PsychRadiometric
	/usr/share/matlab/site/m/psychtoolbox-3/PsychRadiometric/PsychAnsiZ136MPE
	/usr/share/matlab/site/m/psychtoolbox-3/PsychRadiometric/PsychISO2007MPE
	/usr/share/matlab/site/m/psychtoolbox-3/PsychRadiometric/Obsolete
	/usr/share/matlab/site/m/psychtoolbox-3/PsychColorimetricData
	/usr/share/matlab/site/m/psychtoolbox-3/PsychColorimetricData/PsychColorimetricMatFiles
	/usr/share/matlab/site/m/psychtoolbox-3/PsychTests
	/usr/share/matlab/site/m/psychtoolbox-3/PsychTests/TestConfigurations
	/usr/share/matlab/site/m/psychtoolbox-3/PsychTests/UnitTests
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic/Octave3LinuxFilesARM
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic/Octave4WindowsFiles
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic/Octave4LinuxFiles64
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic/Octave4LinuxFiles
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic/Octave3OSXFiles64
	/usr/share/matlab/site/m/psychtoolbox-3/PsychBasic/PsychPlugins
	/usr/share/matlab/site/m/psychtoolbox-3/PsychFiles
	/usr/share/matlab/site/m/psychtoolbox-3/PsychObsolete
	/usr/share/matlab/site/m/psychtoolbox-3/PsychPriority
	/usr/share/matlab/site/m/psychtoolbox-3/PsychRects
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/ExampleCodeForC
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/PsychGLSLShaders
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/MOGL
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/MOGL/source
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/MOGL/source/headers
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/MOGL/source/GL
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/MOGL/core
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/MOGL/wrap
	/usr/share/matlab/site/m/psychtoolbox-3/PsychOpenGL/PsychGLEyetracking
	/usr/share/matlab/site/m/psychtoolbox-3/PsychGLImageProcessing
[...]

**Comment by kleinerm** [comment]
Check if your /usr/local/MATLAB/R2012b/bin/glnxa64/ folder contains conflicting freetype or fontconfig libraries and if so, delete or rename/move them away. Also run "sudo dpkg --reconfigure matlab-support" if you installed Matlab after PTB, or if you didn't answer yes when it asked you if it should rename the Matlab provided libstdc++ or libc or something like that.

It's essentially always an incompatible and outdated libfreetype/libfontconfig or libstdc++ shipped with Matlab, overriding the system libraries.

**Comment by hanneshberlin** [comment]
Thank you very much for your very quick reply and your help, it solved my problem :)

I did not know how to identify conflicting libraries, so I started with trying to reconfigure MATLAB using the command (which I had to change to) "sudo dpkg-reconfigure matlab-support" and choosing YES when asked for renaming.

That did it already, now I can display text again using PTB. Thank you so much!!! I was already really frustrated being stuck there all day not being able to solve the problem... You saved my day!
